### PR TITLE
CB-21680 modify credential validation to allow role arn length between 20 and 2048 chars

### PR DESCRIPTION
**Pull request by biharitomi**

### CODE REVIEW

**Review by doktoric — APPROVED** [review]

File: environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        Assertions.assertTrue(constraintViolations
+                .stream()
+                .anyMatch(cv ->
+                        RoleBasedParameters.ROLE_ARN_LENGTH_VALIDATION_ERROR_MSG.equals(cv.getMessage())
+                ));
+    }
+
+    @Test
+    void testNameValidationWhenNameIsValidWithBasicLength() {
+        String tooShortRoleArn = "arn:aws-us-gov:iam::000000000000:role/cdpe2e-basic";
```
Comment (bergerdenes) [inline environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java:52]: it is not too short anymore :)

File: environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        Assertions.assertFalse(constraintViolations
+                .stream()
+                .anyMatch(cv ->
+                        RoleBasedParameters.ROLE_ARN_LENGTH_VALIDATION_ERROR_MSG.equals(cv.getMessage())
+                ));
+    }
+
+    @Test
+    void testNameValidationWhenNameIsValidWithLengthOver100chars() {
+        String tooShortRoleArn = "arn:aws-us-gov:iam::000000000000:role/cdpe2e-xaccountaccess-stage-xaccountaccess-fxaccountaccess-xaccountaccess";
```
Comment (bergerdenes) [inline environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java:64]: it is not too short anymore :)

**Review by bergerdenes — APPROVED** [review]
Minor

File: environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java (hunk 1 above)
Comment (lajosrodek) [reply environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java:52]: +1

File: environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java (hunk 2 above)
Comment (lajosrodek) [reply environment-api/src/test/java/com/sequenceiq/environment/api/v1/credential/model/parameters/aws/RoleBasedParametersRoleArnLengthTest.java:64]: +1

**Review by lajosrodek — APPROVED** [review]
LGTM

**Pull request merged**